.github/workflows/auto-updates.yaml

```diff
@@ -102,3 +104,51 @@ jobs:
         if: ${{ env.modified == 'true' }}
         run: |
           git push origin auto-update-readme-cli-ref:main
+
+  update-vendored-sources:
+    name: Update XS-Vendored-Sources-Rust on debian/control
+    runs-on: ubuntu-latest
+    container: 
+      image: ubuntu:23.04
+    steps:
+      - name: Install dependencies
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt update
+          DEBIAN_FRONTEND=noninteractive apt install -y cargo dh-cargo git
+      - uses: actions/checkout@v3
+        with:
+          # TODO: Change this when merging it into main
+          ref: user-mount-handler
+      - name: Vendoring the dependencies
+        run: |
+          cargo vendor vendor_rust/
+      - name: Update XS-Vendored-Sources-Rust
+        run: |
+          OUTPUT=$((CARGO_VENDOR_DIR=vendor_rust/ /usr/share/cargo/bin/dh-cargo-vendored-sources 2>&1 ||:) | grep XS-Vendored-Sources-Rust:)
+          if [ -z "$OUTPUT" ]; then
+              exit 0
+          fi
+
+          sed -i "s/^XS-Vendored-Sources-Rust:.*/$OUTPUT/" debian/control
+
+          echo "modified=true" >> $GITHUB_ENV
+        shell: bash
+
+      - run: git config --global --add safe.directory /__w/adsys/adsys
```

Comment:
I don't know if this line is removable... The checkout action changes the directory and then the pull request one gets confused. So I don't know if this will be removable even after the fix

Reply:
What I meant is to add something like this (as it’s the content in qa.yaml):
https://github.com/ubuntu/adsys/blob/main/.github/workflows/qa.yaml#L25

That way, we have a common documentation on what we want.

Reply:
Oh... Got it.